src/routing_message_filter.rs

```diff
@@ -70,9 +72,15 @@ impl RoutingMessageFilter {
 
     // Filter outgoing `RoutingMessage`. Return whether this specific message has been seen recently
     // (and thus should not be sent, due to deduplication).
+    //
+    // Return `false` if serialisation of the message fails - that can be handled elsewhere.
```

Comment:
Hmm, it's complicated... The previous comment said nothing about the case where the message is too large to be serialised, and the behaviour of `big_endian_sip_hash` was that messages that failed to serialise would be treated as having content `vec![]`, and thus all have the same hash (see: https://github.com/maidsafe/maidsafe_utilities/blob/master/src/big_endian_sip_hash.rs#L26-L32). This would mean the *first* oversized message to be inserted would return `false`, and all subsequent ones would return `true`.

We could return to similar behaviour by returning `true` for oversized messages (as in, don't send!), although in the case of the first oversized message, our return value would be different from the current behaviour.

Or, maybe we should defer handling of the error entirely and return a `Result<bool, ...>` that can be handled by the caller (which is almost always another function returning a `Result`, so we can just use `?`).

On the other hand, I don't think it's too significant what we return, because a filtering is option usually only occurs right before or right after a message is serialised for sending (and we handle that failure). The only slightly complicated case I came across is in `relay_to_client`, which would previously have returned `Ok(())` for all oversized messages without sending them. However, I'm pretty sure getting an oversized `RoutingMessage` here is impossible, as its pulled out of a `SignedMessage` and a `SignedMessage` ensures that the contents are serialisable...